changelog/4321.removal.rst

```diff
@@ -1,2 +1,2 @@
-Removed `~sunpy.net.vso.attrs.Source` and `~sunpy.net.vso.attrs.Provider`. They are now `~sunpy.net.attrs.Source` and `~sunpy.net.attrs.Provider` respectively.
-Thus ``attrs``, ``Source`` and ``Provider`` are moved from VSO namespace to `~sunpy.net.attrs`.
+Removed `sunpy.net.vso.attrs.Source` and `sunpy.net.vso.attrs.Provider`. They are now `sunpy.net.attrs.Source` and `sunpy.net.attrs.Provider` respectively.
```

Comment:
but a user can still do `attrs.vso.Source` right? (its just depreciated)

Reply:
Yeah, which means this has the wrong type.

Reply:
oh this is actually the right type.